### PR TITLE
test(bamboo): add status and result tests in e2e

### PR DESCRIPTION
**Pull request by d4x1**
### ⚠️ Pre Checklist

> Please complete _ALL_ items in this checklist, and remove before submitting

- [x] I have read through the [Contributing Documentation](https://devlake.apache.org/community/).
- [x] I have added relevant tests.
- [x] I have added relevant documentation.
- [x] I will add labels to the PR, such as `pr-type/bug-fix`, `pr-type/feature-development`, etc.

### Summary
What does this PR do?
Update Bamboo e2e test, cover status and result fields's transformation.

### Does this close any open issues?
Closes N/A

### Screenshots
Include any relevant screenshots here.
##### cicd_deployment_commits
image
##### cicd_deployments
image
##### cicd_pipelines
image
##### cicd_tasks
image

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

File: backend/plugins/bamboo/e2e/snapshot_tables/cicd_deployment_commits.csv [hunk 1]
```diff
@@ -1,8 +1,15 @@
...
-bamboo:deployBuildWithVcsRevision:1:1540117:622595,79b062bd53af15c701193c90b543386557cb7a3a,bamboo:BambooPlan:1:TEST-PLA2,bamboo:deployBuildWithVcsRevision:1:1540117:622595,release-3,SUCCESS,DONE,dev,2023-08-03T09:49:07.000+00:00,2023-08-03T09:49:07.000+00:00,2023-08-03T09:49:07.000+00:00,0,,622595,fake://127.0.0.1:8080/repos/622595,
+bamboo:deployBuildWithVcsRevision:1:130001:622595,79b062bd53af15c701193c90b543386557cb7a3a,bamboo:BambooPlan:1:TEST-PLA2,bamboo:deployBuildWithVcsRevision:1:130001:622595,test_project2 - test_plan/release-1,SUCCESS,DONE,dev,2023-07-31T10:16:41.000+00:00,2023-07-31T10:16:41.000+00:00,2023-07-31T10:16:41.000+00:00,0,,622595,fake://127.0.0.1:8080/repos/622595,
+bamboo:deployBuildWithVcsRevision:1:130002:622595,79b062bd53af15c701193c90b543386557cb7a3a,bamboo:BambooPlan:1:TEST-PLA2,bamboo:deployBuildWithVcsRevision:1:130002:622595,test_project2 - test_plan/release-1,FAILURE,IN_PROGRESS,dev,2023-07-31T10:16:41.000+00:00,2023-07-31T10:16:41.000+00:00,2023-07-31T10:16:41.000+00:00,0,,622595,fake://127.0.0.1:8080/repos/622595,
```
Comment (abeizn) [inline backend/plugins/bamboo/e2e/snapshot_tables/cicd_deployment_commits.csv:3]: FAILURE,IN_PROGRESS ?

File: backend/plugins/bamboo/e2e/snapshot_tables/cicd_deployments.csv [hunk 2]
```diff
@@ -0,0 +1,15 @@
+id,commit_sha,cicd_scope_id,cicd_deployment_id,name,result,status,environment,created_date,started_date,finished_date,duration_sec,ref_name,repo_id,repo_url,prev_success_deployment_commit_id
+bamboo:deployBuildWithVcsRevision:1:130001:622595,79b062bd53af15c701193c90b543386557cb7a3a,bamboo:BambooPlan:1:TEST-PLA2,bamboo:deployBuildWithVcsRevision:1:130001:622595,test_project2 - test_plan/release-1,SUCCESS,DONE,dev,2023-07-31T10:16:41.000+00:00,2023-07-31T10:16:41.000+00:00,2023-07-31T10:16:41.000+00:00,0,,622595,fake://127.0.0.1:8080/repos/622595,
+bamboo:deployBuildWithVcsRevision:1:130002:622595,79b062bd53af15c701193c90b543386557cb7a3a,bamboo:BambooPlan:1:TEST-PLA2,bamboo:deployBuildWithVcsRevision:1:130002:622595,test_project2 - test_plan/release-1,FAILURE,IN_PROGRESS,dev,2023-07-31T10:16:41.000+00:00,2023-07-31T10:16:41.000+00:00,2023-07-31T10:16:41.000+00:00,0,,622595,fake://127.0.0.1:8080/repos/622595,
```
Comment (abeizn) [inline backend/plugins/bamboo/e2e/snapshot_tables/cicd_deployments.csv:3]: same as above

File: backend/plugins/bamboo/e2e/snapshot_tables/cicd_pipelines.csv [hunk 3]
```diff
@@ -10,3 +10,8 @@ bamboo:BambooPlanBuild:1:TEST-PLA2-6,test - pla-2/6,FAILURE,DONE,,0,,2023-07-31T
 bamboo:BambooPlanBuild:1:TEST-PLA2-7,test - pla-2/7,FAILURE,DONE,,0,,2023-07-31T09:17:05.252+00:00,2023-07-31T09:17:05.507+00:00,bamboo:BambooPlan:1:TEST-PLA2
 bamboo:BambooPlanBuild:1:TEST-PLA2-8,test - pla-2/8,FAILURE,DONE,,0,,2023-07-31T09:17:35.396+00:00,2023-07-31T09:17:35.654+00:00,bamboo:BambooPlan:1:TEST-PLA2
 bamboo:BambooPlanBuild:1:TEST-PLA2-9,test - pla-2/9,FAILURE,DONE,,0,,2023-07-31T09:20:13.276+00:00,2023-07-31T09:20:13.609+00:00,bamboo:BambooPlan:1:TEST-PLA2
+bamboo:BambooPlanBuild:1:TEST-PLA3-10001,test - pla-3/1,SUCCESS,DONE,,1,,2023-07-27T03:47:12.941+00:00,2023-07-27T03:47:14.079+00:00,bamboo:BambooPlan:1:TEST-PLA2
+bamboo:BambooPlanBuild:1:TEST-PLA3-10002,test - pla-3/1,SUCCESS,IN_PROGRESS,,1,,2023-07-27T03:47:12.941+00:00,2023-07-27T03:47:14.079+00:00,bamboo:BambooPlan:1:TEST-PLA2
```
Comment (abeizn) [inline backend/plugins/bamboo/e2e/snapshot_tables/cicd_pipelines.csv:14]: SUCCESS,IN_PROGRESS?

File: backend/plugins/bamboo/e2e/snapshot_tables/cicd_pipelines.csv [hunk 4]
```diff
@@ -10,3 +10,8 @@ bamboo:BambooPlanBuild:1:TEST-PLA2-6,test - pla-2/6,FAILURE,DONE,,0,,2023-07-31T
...
+bamboo:BambooPlanBuild:1:TEST-PLA3-10003,test - pla-3/1,FAILURE,IN_PROGRESS,,1,,2023-07-27T03:47:12.941+00:00,2023-07-27T03:47:14.079+00:00,bamboo:BambooPlan:1:TEST-PLA2
```
Comment (abeizn) [inline backend/plugins/bamboo/e2e/snapshot_tables/cicd_pipelines.csv:15]: same

File: backend/plugins/bamboo/e2e/snapshot_tables/cicd_pipelines.csv [hunk 5]
```diff
@@ -10,3 +10,8 @@ bamboo:BambooPlanBuild:1:TEST-PLA2-6,test - pla-2/6,FAILURE,DONE,,0,,2023-07-31T
...
+bamboo:BambooPlanBuild:1:TEST-PLA3-10004,test - pla-3/1,FAILURE,IN_PROGRESS,,1,,2023-07-27T03:47:12.941+00:00,2023-07-27T03:47:14.079+00:00,bamboo:BambooPlan:1:TEST-PLA2
```
Comment (abeizn) [inline backend/plugins/bamboo/e2e/snapshot_tables/cicd_pipelines.csv:16]: same

File: backend/plugins/bamboo/e2e/snapshot_tables/cicd_tasks.csv [hunk 6]
```diff
@@ -1,3 +1,8 @@
 id,name,pipeline_id,result,status,type,environment,duration_sec,started_date,finished_date,cicd_scope_id
 bamboo:BambooJobBuild:1:TEST-PLA3-JOB1-1,Default Job,bamboo:BambooPlanBuild:1:TEST-PLA3-1,SUCCESS,DONE,,,0,2023-07-27T03:47:13.626+00:00,2023-07-27T03:47:13.750+00:00,bamboo:BambooPlan:1:TEST-PLA3
 bamboo:BambooJobBuild:1:TEST-PLA3-JOB1-2,Default Job,bamboo:BambooPlanBuild:1:TEST-PLA3-2,SUCCESS,DONE,,,0,2023-07-27T03:56:15.783+00:00,2023-07-27T03:56:15.789+00:00,bamboo:BambooPlan:1:TEST-PLA3
+bamboo:BambooJobBuild:1:TEST-PLA3-JOB1-20001,Default Job,bamboo:BambooPlanBuild:1:TEST-PLA3-1,SUCCESS,DONE,,,0,2023-07-27T03:47:13.626+00:00,2023-07-27T03:47:13.750+00:00,bamboo:BambooPlan:1:TEST-PLA3
+bamboo:BambooJobBuild:1:TEST-PLA3-JOB1-20002,Default Job,bamboo:BambooPlanBuild:1:TEST-PLA3-1,SUCCESS,IN_PROGRESS,,,0,2023-07-27T03:47:13.626+00:00,2023-07-27T03:47:13.750+00:00,bamboo:BambooPlan:1:TEST-PLA3
```
Comment (abeizn) [inline backend/plugins/bamboo/e2e/snapshot_tables/cicd_tasks.csv:5]: same

File: backend/plugins/bamboo/e2e/snapshot_tables/cicd_tasks.csv [hunk 7]
```diff
@@ -1,3 +1,8 @@
...
+bamboo:BambooJobBuild:1:TEST-PLA3-JOB1-20003,Default Job,bamboo:BambooPlanBuild:1:TEST-PLA3-1,SUCCESS,IN_PROGRESS,,,0,2023-07-27T03:47:13.626+00:00,2023-07-27T03:47:13.750+00:00,bamboo:BambooPlan:1:TEST-PLA3
```
Comment (abeizn) [inline backend/plugins/bamboo/e2e/snapshot_tables/cicd_tasks.csv:6]: same

File: backend/plugins/bamboo/e2e/snapshot_tables/cicd_tasks.csv [hunk 8]
```diff
@@ -1,3 +1,8 @@
...
+bamboo:BambooJobBuild:1:TEST-PLA3-JOB1-20004,Default Job,bamboo:BambooPlanBuild:1:TEST-PLA3-1,FAILURE,IN_PROGRESS,,,0,2023-07-27T03:47:13.626+00:00,2023-07-27T03:47:13.750+00:00,bamboo:BambooPlan:1:TEST-PLA3
```
Comment (abeizn) [inline backend/plugins/bamboo/e2e/snapshot_tables/cicd_tasks.csv:7]: same

File: backend/plugins/bamboo/tasks/deploy_build_convertor.go [hunk 9]
```diff
@@ -94,6 +94,7 @@ func ConvertDeployBuilds(taskCtx plugin.SubTaskContext) errors.Error {
 	}
 	defer cursor.Close()
 
+	fmt.Println("aaa", data.Options.PlanKey, data.Options.ConnectionId)
```
Comment (abeizn) [inline backend/plugins/bamboo/tasks/deploy_build_convertor.go:97]: remove it

File: backend/plugins/bamboo/e2e/snapshot_tables/cicd_deployment_commits.csv (hunk 1 above)
Comment (d4x1) [reply backend/plugins/bamboo/e2e/snapshot_tables/cicd_deployment_commits.csv:3]: These data are fake, result and status fields are transformed seperately. So it's normal that one record can have  `FAILURE` result and `IN_PROGRESS` status. @abeizn

File: backend/plugins/bamboo/tasks/deploy_build_convertor.go (hunk 9 above)
Comment (d4x1) [reply backend/plugins/bamboo/tasks/deploy_build_convertor.go]: Got it.

**Review by abeizn — APPROVED** [review]

**Pull request merged**